### PR TITLE
game_logic

### PR DESCRIPTION
**Pull request by SamanAtashi**
# MileStone 3: Game logic for Tic Tac Toe game
* Basically, the `win?` methods are added to determine the winner.

### CODE REVIEW

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi Saman!
Great job on this project:clap::clap:.** 
**Major highlights**

- Thank you for providing a very descriptive readme.
- The codebase is clean, easy to read and understand.

However, we still need to make a few changes to get this project approved:muscle:.

**To improve**

- [x] Board.rb

- Refactor the `printing` method in line 15 - 25 to move all the `print` statements to the main.rb.  No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [x] human_player.rb

- Refactor `get_position` method so as to move line  13 to main.rb. No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)).

- [x] Implementation details

- All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). Move the board.rb and human_player.rb to the lib directory.

Make the above-required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  def empty?(pos)
+    %w[1 2 3 4 5 6 7 8 9].include?(@grid[pos[0]][pos[1]])
+  end
+
+  def place_mark(position, mark)
+    if valid?(position) && empty?(position)
+      @grid[position[0]][position[1]] = mark
+    else
+      puts 'Invalid mark'
+      puts 'Try again'
```
Comment (rloterh) [inline lib/board.rb]: - [x] It is a good practice to place all `puts`, `print` and `chomps` in the main. Kindly ensure to move the `puts` into the `main.rb`

**Review by rloterh — CHANGES_REQUESTED** [review]
# Changes Required! ♻️
Hi @samanatashi!👋 👋
Great job for implementing most changes required by the previous reviewer(s). 👏  You're almost there, however,  a bit more refining is required to improve your code.

Please check the comments below for the necessary changes needed to be improved


Keep working harder 🍷 💻. 
Feel free to drop a comment in case of any inquiries regarding this review. 😉

__Robert Loterh__

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @SamanAtashi ! :wave: :wave:

####  Good job for making all the necessary corrections recommended by the previous reviewer(s). 👏 . Your code looks good,  👍  Congratulations!  :clap:


#### You may merge, good luck with the next milestone! :rocket:

If you have any questions please feel free to drop me a comment.

#### Happy Coding! :+1:

__Robert Loterh __
[GitHub](https://github.com/rloterh) | [LinkedIn](https://www.linkedin.com/in/rloterh/) | [Twitter](https://twitter.com/RLoterh)

**Pull request merged**